google-cloud-spanner/lib/google/cloud/spanner/admin/database.rb

```diff
@@ -0,0 +1,209 @@
+            else
+              credentials ||= default_credentials scope: scope
+              unless credentials.is_a? Google::Auth::Credentials
+                credentials = Spanner::Credentials.new credentials, scope: scope
+              end
+
+              if credentials.respond_to? :project_id
+                project_id ||= credentials.project_id
+              end
+            end
+
+            project_id = project_id.to_s # Always cast to a string
+            raise ArgumentError, "project_id is missing" if project_id.empty?
+
+            Admin::Database::V1::DatabaseAdmin::Client.new do |config|
+              config.credentials = channel endpoint, credentials
+              config.timeout = timeout if timeout
+              config.endpoint = endpoint if endpoint
+              config.lib_name = lib_name_with_prefix lib_name, lib_version
+              config.lib_version = Google::Cloud::Spanner::VERSION
+              config.metadata = { "google-cloud-resource-prefix" => "projects/#{project_id}" }
+            end
+          end
+
+          ##
+          # Configure the Google Cloud Spanner Database Admin library.
+          #
+          # The following configuration parameters are supported:
+          #
+          # * `credentials` (*type:* `String, Hash, Google::Auth::Credentials`) -
+          #   The path to the keyfile as a String, the contents of the keyfile as a
+          #   Hash, or a Google::Auth::Credentials object.
+          # * `lib_name` (*type:* `String`) -
+          #   The library name as recorded in instrumentation and logging.
+          # * `lib_version` (*type:* `String`) -
+          #   The library version as recorded in instrumentation and logging.
+          # * `interceptors` (*type:* `Array<GRPC::ClientInterceptor>`) -
+          #   An array of interceptors that are run before calls are executed.
+          # * `timeout` (*type:* `Numeric`) -
+          #   Default timeout in seconds.
+          # * `emulator_host` - (String) Host name of the emulator. Defaults to
+          #   `ENV["SPANNER_EMULATOR_HOST"]`.
+          # * `metadata` (*type:* `Hash{Symbol=>String}`) -
+          #   Additional gRPC headers to be sent with the call.
+          # * `retry_policy` (*type:* `Hash`) -
+          #   The retry policy. The value is a hash with the following keys:
+          #     * `:initial_delay` (*type:* `Numeric`) - The initial delay in seconds.
+          #     * `:max_delay` (*type:* `Numeric`) - The max delay in seconds.
+          #     * `:multiplier` (*type:* `Numeric`) - The incremental backoff multiplier.
+          #     * `:retry_codes` (*type:* `Array<String>`) -
+          #       The error codes that should trigger a retry.
+          #
+          # @return [::Google::Cloud::Config] The default configuration used by this library
+          #
+          def self.configure
+            yield ::Google::Cloud.configure.spanner if block_given?
```

Comment:
I think the configuration should be the same for the three clients though if we're trying to keep things more or less the same as before.

What are the cases you were thinking of where it might be confusing for users? E.g. they try to have a different timeout for admin clients compared to the data client?

Where I think it's useful to use the same configuration is when using the emulator for example.